File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples-2012/CSharp/WPFListBox/WPFComponents/Converters/XmlElementToDecimalConverter.cs
using System;
using System.ComponentModel;
using System.Globalization;
using System.Windows.Data;
using System.Xml;

namespace WPFComponents
{
  [ValueConversion(typeof(XmlElement), typeof(decimal))]
  public class XmlElementToDecimalConverter : IValueConverter
  {
    public object Convert(object value, Type targetType,
                                        object parameter, CultureInfo culture)
    {
      return System.Convert.ToDecimal(((XmlElement)value).InnerText);
    }

    public object ConvertBack(object value, Type targetType, 
                                                object parameter, CultureInfo culture)
    {
      return System.Convert.ToString(value);
    }
  }
}

File: _wpf/wpf-business-programmer/2-wpf-business-programmer-m2-exercise-files/Samples/VB/WPFListBox-VB/WPFComponents/Converters/XmlElementToDecimalConverter.vb
﻿Imports System.Globalization
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data
Imports System.Xml

<ValueConversion(GetType(XmlElement), GetType(Decimal))> _
Public Class XmlElementToDecimalConverter
  Implements IValueConverter

  Public Function Convert(ByVal value As Object, _
                          ByVal targetType As Type, _
                          ByVal parameter As Object, _
                          ByVal culture As CultureInfo) As Object _
                  Implements System.Windows.Data.IValueConverter.Convert
    Return System.Convert.ToDecimal(DirectCast(value, XmlElement).InnerText)
  End Function

  Public Function ConvertBack(ByVal value As Object, _
                              ByVal targetType As Type, _
                              ByVal parameter As Object, _
                              ByVal culture As CultureInfo) As Object _
                              Implements System.Windows.Data.IValueConverter.ConvertBack
    Return System.Convert.ToString(value)
  End Function
End Class